ngtcp2/Dockerfile-client

```diff
@@ -1,59 +1,71 @@
 
 # build nghttp3
-RUN git clone https://github.com/ngtcp2/nghttp3 && cd nghttp3 && autoreconf -i && ./configure --prefix=$PWD/build --enable-lib-only && make -j$(nproc) check && make install
-
-# build ngtcp2
-RUN git clone https://github.com/ngtcp2/ngtcp2 && cd ngtcp2 && autoreconf -i && ./configure PKG_CONFIG_PATH=$PWD/../oqs-openssl-quic/build/lib/pkgconfig:$PWD/../nghttp3/build/lib/pkgconfig LDFLAGS="-Wl,-rpath,$PWD/../oqs-openssl-quic/build/lib"  CXX=g++-11 && make -j$(nproc) check
-
+WORKDIR /opt/nghttp3
+RUN autoreconf -i && ./configure --prefix=$PWD/build --enable-lib-only && make -j$(nproc) check && make install
 
-# Copy executable to usr/local/bin
-RUN cp ngtcp2/examples/client /usr/local/bin/ 
 
-# quicreach
-RUN git config --global protocol.file.allow always
-RUN git clone --recursive https://github.com/microsoft/quicreach && cd quicreach/msquic && git submodule deinit -f -- submodules/openssl && rm -rf ../.git/modules/msquic/modules/submodules/openssl && git rm -f submodules/openssl && git submodule add /opt/oqs-openssl-quic submodules/openssl && git submodule update --init submodules/googletest && cd ../..
-
-
-RUN cd liboqs && cd build && cmake -GNinja -DCMAKE_INSTALL_PREFIX=/opt/quicreach/msquic/submodules/openssl/oqs .. && ninja && ninja install && cp -R /opt/quicreach/msquic/submodules/openssl/oqs/include/oqs /opt/quicreach/msquic/submodules/openssl/include
-
-COPY CMakeLists.txt.patch /opt/quicreach/msquic
-RUN patch /opt/quicreach/msquic/CMakeLists.txt < /opt/quicreach/msquic/CMakeLists.txt.patch
-COPY src-inc-CMakeLists.txt.patch /opt/quicreach/msquic
-RUN patch /opt/quicreach/msquic/src/inc/CMakeLists.txt < /opt/quicreach/msquic/src-inc-CMakeLists.txt.patch
+# build ngtcp2
+WORKDIR /opt/ngtcp2
+RUN autoreconf -i && ./configure PKG_CONFIG_PATH=$PWD/../nghttp3/build/lib/pkgconfig:/opt/oqssa/lib64/pkgconfig CXX=g++ CXXFLAGS="-std=c++20" && make -j$(nproc) && make install
+
+# Copy all required shared object dependencies to a single directory
+RUN mkdir /opt/lib && cd /opt/lib && \
+        cp /opt/ngtcp2/lib/.libs/libngtcp2.so.11 . && \
+        cp /usr/lib/libev.so.4 . && \
+        cp /opt/nghttp3/build/lib/libnghttp3.so.5 . && \
+        cp /opt/ngtcp2/crypto/openssl/.libs/libngtcp2_crypto_openssl.so.4 . && \
+        cp /opt/oqssa/lib64/libssl.so.81.3 . && \
+        cp /opt/oqssa/lib64/libcrypto.so.81.3 . && \
+        cp /usr/lib/libstdc++.so.6 . && \
```

Comment:
These `cp` commands look "brittle": What if any of the library versions change? The docker build will not succeed any more...

Reply:
Thank you for the feedback! I changed them to use wildcards instead.